dotenv/src/parse.rs

```diff
@@ -1,63 +1,57 @@
 use std::collections::HashMap;
 
-use lazy_static::lazy_static;
-use regex::{Captures, Regex};
-
 use crate::errors::*;
 
 // for readability's sake
 pub type ParsedLine = Result<Option<(String, String)>>;
 
-pub fn parse_line(line: &str, mut substitution_data: &mut HashMap<String, Option<String>>) -> ParsedLine {
-    lazy_static! {
-      static ref LINE_REGEX: Regex = Regex::new(r#"(?x)
-        ^(
-          \s*
-          (
-            \#.*|                             # A comment, or...
-            \s*|                              # ...an empty string, or...
-            (export\s+)?                      # ...(optionally preceded by "export")...
-            (?P<key>[A-Za-z_][A-Za-z0-9_.]*)  # ...a key,...
-            =                                 # ...then an equal sign,...
-            (?P<value>.+?)?                   # ...and then its corresponding value.
-          )\s*
-        )
-        [\r\n]*
-        $
-      "#).unwrap();
+pub fn parse_line(
+    original_line: &str,
+    mut substitution_data: &mut HashMap<String, Option<String>>,
+) -> ParsedLine {
+    // trim whitespace
+    let mut line = original_line.trim();
+
+    // if its an empty line or a comment, skip it
+    if line.is_empty() || line.starts_with('#') {
+        return Ok(None);
     }
 
-    LINE_REGEX
-        .captures(line)
-        .map_or(Err(Error::LineParse(line.into(), 0)), |captures| {
-            let key = named_string(&captures, "key");
-            let value = named_string(&captures, "value");
+    // optionally preceded by `export\s+`
+    if line.starts_with("export") {
+        let without_export = line.trim_start_matches("export");
+        if !without_export.starts_with(|c: char| c.is_whitespace()) {
+            return Err(Error::LineParse(original_line.into(), 0));
+        }
+        line = without_export.trim_start();
+    }
 
-            match (key, value) {
-                (Some(k), Some(v)) => {
-                    let parsed_value = parse_value(&v, &mut substitution_data)?;
-                    substitution_data.insert(k.to_owned(), Some(parsed_value.to_owned()));
+    // parse key
+    if !line.starts_with(|c: char| c.is_ascii_alphabetic() || c == '_') {
+        return Err(Error::LineParse(original_line.into(), 0));
+    }
+    let key: String = line
+        .chars()
+        .take_while(|c| c.is_ascii_alphanumeric() || *c == '_' || *c == '.')
+        .collect();
+    line = &line[key.len()..];
+
+    // then ensure the equal sign
+    if !line.starts_with('=') {
```

Comment:
Trim leading whitespace here, that way we can support spaces after the key and before the equals sign.